<?xml version="1.0" encoding="utf-8"?>	
	
<launch>


  <rosparam command="load" file="$(find cone_detection)/cfg/parameter.yaml" />

  <!-- CONE DETECTION -->
  <node name="cone_detector" pkg="cone_detection" type="cone_detector" output="screen" />
  <node name="cone_eval_net" pkg="cone_detection" type="cone_eval_net.py" output="screen" />
  <node name="viewer" pkg="cone_detection" type="viewer.py" output="screen" />
  <node name="path_planner" pkg="cone_detection" type="path_planner" output="screen" />

  <!-- WEBCAM -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" />
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="false" />
  </node>

  <!-- VELODYNE -->
  	<arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
  	<arg name="device_ip" default="" />
  	<arg name="frame_id" default="velodyne" />
  	<arg name="manager" default="$(arg frame_id)_nodelet_manager" />
  	<arg name="max_range" default="130.0" />
  	<arg name="min_range" default="0.4" />
  	<arg name="pcap" default="" />
  	<arg name="port" default="2368" />
  	<arg name="read_fast" default="false" />
  	<arg name="read_once" default="false" />
  	<arg name="repeat_delay" default="0.0" />
  	<arg name="rpm" default="600.0" />
  <include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
    	<arg name="device_ip" value="$(arg device_ip)"/>
    	<arg name="frame_id" value="$(arg frame_id)"/>
    	<arg name="manager" value="$(arg manager)" />
    	<arg name="model" value="VLP16"/>
    	<arg name="pcap" value="$(arg pcap)"/>
    	<arg name="port" value="$(arg port)"/>
    	<arg name="read_fast" value="$(arg read_fast)"/>
    	<arg name="read_once" value="$(arg read_once)"/>
    	<arg name="repeat_delay" value="$(arg repeat_delay)"/>
   	<arg name="rpm" value="$(arg rpm)"/>
  </include>
  <include file="$(find velodyne_pointcloud)/launch/cloud_nodelet.launch">
   	<arg name="calibration" value="$(arg calibration)"/>
    	<arg name="manager" value="$(arg manager)" />
    	<arg name="max_range" value="$(arg max_range)"/>
    	<arg name="min_range" value="$(arg min_range)"/>
  </include>
  
  <!-- VI -->
  <node name="visensor_node" pkg="visensor_node" type="visensor_node" output="screen"/>

  <!-- ROVIO -->
  <node pkg="rovio" type="rovio_node" name="rovio" output="screen">
   <param name="filter_config" value="$(find rovio)/cfg/arc_old.info"/>
   <param name="camera0_config" value="$(find rovio)/cfg/rovio_cam0.yaml"/>
   <param name="camera1_config" value="$(find rovio)/cfg/rovio_cam1.yaml"/>
  <param name="use_car_model_as_odom" value="false"/>
  </node> 


</launch>

